src/Microsoft.Repl.Tests/ShellTests.cs

```diff
@@ -0,0 +1,191 @@
+using System;
+using System.Threading;
+using Microsoft.HttpRepl;
+using Microsoft.Repl.Commanding;
+using Microsoft.Repl.ConsoleHandling;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Repl.Tests
+{
+    public class ShellTests
+    {
+        [Fact]
+        public void RunAsync_WithUpArrowKeyPress_UpdatesCurrentBufferWithPreviousCommand()
+        {
+            string previousCommand = "set base \"https://localhost:44366/\"";
+            Shell shell = CreateShell(ConsoleKey.UpArrow,
+                0,
+                previousCommand,
+                null,
+                out CancellationTokenSource cancellationTokenSource);
+
+            // Verify the input buffer is empty before the UpArrow key press event
+            Assert.Equal(string.Empty, shell.ShellState.InputManager.GetCurrentBuffer());
```

Comment:
This isn't relevant to the test scenario, is it?

Reply:
I want to verify buffer contents before and after keypress events.

Reply:
After makes sense, but in this test case, it's irrelevant what it is before, right?  

Actually, now that I think about it, for thoroughness you could have a test where there is something on the buffer before and verify that pressing Up/Down replaces it.

Reply:
So removed before assert from all tests and added new test to check buffer contents before and after key press event:
https://github.com/aspnet/HttpRepl/commit/2f322167e964e7f08a6b8343ea0943296f22ad84